pyccel/codegen/printing/cucode.py

```diff
@@ -52,23 +52,42 @@ def _print_Module(self, expr):
 
         # Print imports last to be sure that all additional_imports have been collected
         imports = [Import(expr.name, Module(expr.name,(),())), *self._additional_imports.values()]
-        c_headers_imports = ''
-        local_imports = ''
-
-        for imp in imports:
-            if imp.source in c_library_headers:
-                c_headers_imports += self._print(imp)
-            else:
-                local_imports += self._print(imp)
-
-        imports = f'{c_headers_imports}\
-                    extern "C"{{\n\
-                    {local_imports}\
-                    }}'
+        imports = ''.join(self._print(i) for i in imports)
 
         code = f'{imports}\n\
                  {global_variables}\n\
                  {body}\n'
 
         self.exit_scope()
         return code
+
+    def _print_ModuleHeader(self, expr):
+        self.set_scope(expr.module.scope)
+        self._in_header = True
+        name = expr.module.name
+
+        funcs = ""
+        cuda_headers = ""
+        for f in expr.module.funcs:
+            if not f.is_inline:
+                if 'kernel' in f.decorators:  # Checking for 'kernel' decorator
+                    cuda_headers += self.function_signature(f) + ';\n'
+                else:
+                    funcs += self.function_signature(f) + ';\n'
+        global_variables = ''.join('extern '+self._print(d) for d in expr.module.declarations if not d.variable.is_private)
+        # Print imports last to be sure that all additional_imports have been collected
+        imports = [*expr.module.imports, *self._additional_imports.values()]
+        imports = ''.join(self._print(i) for i in imports)
+
+        self._in_header = False
+        self.exit_scope()
+        function_declaration = f'{cuda_headers}\n\
+                    extern "C"{{\n\
+                    {funcs}\
+                    }}'
+        return (f"#ifndef {name.upper()}_H\n \
+                #define {name.upper()}_H \
+                {global_variables}\n \
+                { function_declaration}\n\n \
+                #endif // {name.upper()}_H\n")
```

Comment:
We have tried to make sure that string expressions in Pyccel end with `\n` so we can easily use `join` to combine them in a readable way without needing to worry about indenting. E.g here I would suggest:
```suggestion
        function_declaration = f'{cuda_headers}\n\
                    extern "C"{{\n\
                    {funcs}\
                    }}\n'
        return '\n'.join((f"#ifndef {name.upper()}_H",
                          f"#define {name.upper()}_H",
                          global_variables,
                          function_declaration
                          "#endif // {name.upper()}_H\n"))
```

Reply:
fixed d48de5b

Reply:
Thanks. Please update your comment with the before/after code

Reply:
yeah sure

Reply:
dn

Reply:
There is no big change, as the example doesn't have global variables

Reply:
should I add one more? I ask because I don't think it's necessary.

Reply:
No I don't think it's necessary. It might have been nice to pick a test case that covered all changes but 2 test cases feels like overkill